lib/matplotlib/axes/_axes.py

```diff
@@ -2002,38 +2061,38 @@ def bar(self, left, height, width=0.8, bottom=None, **kwargs):
         label = kwargs.pop('label', '')
         tick_labels = kwargs.pop('tick_label', None)
 
-        _left = left
-        _bottom = bottom
-        left, height, width, bottom = np.broadcast_arrays(
+        _x = x
+        _y = y
+        x, height, width, y, linewidth = np.broadcast_arrays(
             # Make args iterable too.
-            np.atleast_1d(left), height, width, bottom)
+            np.atleast_1d(x), height, width, y, linewidth)
 
         adjust_ylim = False
         adjust_xlim = False
         if orientation == 'vertical':
-            self._process_unit_info(xdata=left, ydata=height, kwargs=kwargs)
+            self._process_unit_info(xdata=x, ydata=height, kwargs=kwargs)
             if log:
                 self.set_yscale('log', nonposy='clip')
-            # size width and bottom according to length of left
-            if _bottom is None:
+            # size width and y according to length of x
+            if _y is None:
```

Comment:
Can we just default bottom to 0 instead of None?  (looks so, except possibly for log scale?)

Reply:
we would have to keep this logic anyway as users can be passing `None` in explicitly.  Inclined to let that stay as-is for now.

Reply:
Can we at least make the change go before the call to np.broadcast_arrays, so that you don't have to call zeros_like below?
(I would also deprecate allowing passing None yada yada)